### PR TITLE
suspicious request blocking tests upgrade

### PR DESCRIPTION
**Pull request by christophe-papazian**
## Description

rework of tests for suspicious request blocking feature for ASM after https://github.com/DataDog/system-tests/pull/1078

- remove set_value/get_value entry points and use a tag_value that should add the value obtained in path parameters in the meta tags of the span
- update tests accordingly
- use urlencoded body for request.body test as this should be supported by all ASM libraries

### CODE REVIEW

**Review by simon-id — COMMENTED** [review]
Can't see anything wrong with it but I have zero python experience so take it with a grain of salt

File: docs/weblog/README.md
Comment (simon-id) [inline docs/weblog/README.md:136]:
```suggestion
The first path parameter must be written in the span with the tag `appsec.events.system_tests_appsec_event.value` and the parameter as value.
```

File: docs/weblog/README.md
Comment (simon-id) [inline docs/weblog/README.md:140]: what response header ?

File: tests/appsec/test_blocking_addresses.py
Comment (Anilm3) [inline tests/appsec/test_blocking_addresses.py:144]: It's unclear to me what the difference between raising an exception and asserting is in this context, but I was wondering if replacing the `if/raise` with `assert value == expected_value`  would have the same effect?

**Review by Anilm3 — APPROVED** [review]
lgtm, not sure if my approval will allow you to merge,but in case it does you should probably wait for @cbeauchesne's approval.

File: tests/appsec/test_blocking_addresses.py
Comment (christophe-papazian) [reply tests/appsec/test_blocking_addresses.py]: I replaced that by an assert as suggested.

File: docs/weblog/README.md
Comment (Julio-Guerra) [inline docs/weblog/README.md:140]: Unclear to me. Maybe provide an example?

Comment (christophe-papazian) [reply docs/weblog/README.md:140]: Yes, good idea, I add a full example soon.

File: tests/appsec/test_blocking_addresses.py
Comment (cbeauchesne) [inline tests/appsec/test_blocking_addresses.py:138]: It's more for the code clarety, could you split this in twwo functions : 

```python
def _assert_custom_event_tag_presence(expected_value):
def _assert_custom_event_tag_absence():
```

IMO, the test line will be a little more clear, and won't require any cognitive effort on reading nested `if/else`.

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/appsec/test_blocking_addresses.py
Comment (christophe-papazian) [reply tests/appsec/test_blocking_addresses.py]: Done

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**
